### PR TITLE
Add join VNET call for every AZR NC unpublish call with fixed UTs

### PR DESCRIPTION
**Pull request by smittal22**
**Reason for Change**:
For AZR scenarios, if NMAgent is restarted, it loses state and does not know what VNETs to subscribe to. As it no longer has VNET state, successive put nc or delete nc calls would fail. We, however, always call join VNET before put nc call. This PR adds such a flow for delete AZR nc calls as well.

PS: We don't need this flow for regular NCs, as NMAgent there uses NMA swift module to rehydrate VNETs if state is lost.

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: cns/restserver/api.go [hunk 1]
```diff
@@ -1029,7 +1031,27 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
...
+		// If the body contains only `""\n`, it is non-AZR NC
+		// In this case, we should not return an error
+		// However, if the body is not `""\n`, it is invalid and therefore, we must return an error
+		if !bytes.Equal(req.DeleteNetworkContainerRequestBody, []byte(`""`+"\n")) {
+			http.Error(w, fmt.Sprintf("could not unmarshal delete network container body: %v", err), http.StatusBadRequest)
+			return
+		}
+	} else {
+		// If unmarshalling was successful, it is an AZR NC
+		azrNC = true
```
Comment (timraymond) [inline cns/restserver/api.go:1047]: This is too much knowledge about NMAgent quirks in a component that shouldn't care. I'd recommend making a `nmagent.NewDeleteContainerRequest` that returns a request and an error. Then all the API needs to care about is whether or not that function says it was able to create a request or not. Then, ideally... you can just ask the request if it's an AZR NC or not instead of needing the other bool for it.

Comment (smittal22) [reply cns/restserver/api.go:1047]: We are not really creating any request. Just passing along the body that CNS received to NMA as payload.

File: cns/restserver/api.go [hunk 2]
```diff
@@ -1029,7 +1031,27 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
...
 
-	if !service.isNetworkJoined(req.NetworkID) {
+	var unpublishBody nmagent.DeleteContainerRequest
+	var azrNC bool
+	err = json.Unmarshal(req.DeleteNetworkContainerRequestBody, &unpublishBody)
+	if err != nil {
+		// If the body contains only `""\n`, it is non-AZR NC
+		// In this case, we should not return an error
+		// However, if the body is not `""\n`, it is invalid and therefore, we must return an error
+		if !bytes.Equal(req.DeleteNetworkContainerRequestBody, []byte(`""`+"\n")) {
```
Comment (ramiro-gamarra) [inline cns/restserver/api.go:1041]: the bytes are hard to parse due to the mix of raw and regular string literals. Easier to just write `[]byte{34, 34, 10}` and explain what it is in the comment?

File: cns/restserver/api_test.go [hunk 3]
```diff
@@ -1068,20 +1068,71 @@ func TestUnpublishNCViaCNS(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			errPublish := publishNCViaCNS(vnet, tt.ncID, createNetworkContainerURL)
+			require.NoError(t, errPublish)
+			errUnpublish := unpublishNCViaCNS(vnet, tt.ncID, deleteNetworkContainerURL, tt.body)
+			if tt.requireError {
+				require.Error(t, errUnpublish)
```
Comment (ramiro-gamarra) [inline cns/restserver/api_test.go:1128]: the error could be coming from multiple places, not necessarily from the bad input from the test. you need a more thorough check on the type of error you expect from this specific bad input.

File: cns/restserver/api.go (hunk 2 above)
Comment (smittal22) [reply cns/restserver/api.go:1041]: Yeah, i had a tossup between the two and thought visually the bytes make more sense, but I can change that up.

Comment (smittal22) [reply cns/restserver/api.go:1041]: Changed

File: cns/restserver/api_test.go (hunk 3 above)
Comment (smittal22) [reply cns/restserver/api_test.go:1128]: addressed

**Review by ramiro-gamarra — APPROVED** [review]
